convert2rhel/repo.py

```diff
@@ -169,3 +170,25 @@ def _restore_varsdir(path):
     if system_info.version.major == 8:
         loggerinst.task("Rollback: Restore variable files to %s", DEFAULT_DNF_VARS_DIR)
         _restore_varsdir(DEFAULT_DNF_VARS_DIR)
+
+
+def clean_yum_metadata():
+    """Remove cached metadata from yum.
+
+    This is to make sure that Convert2RHEL works with up-to-date data from repositories before, for instance, querying
+    whether the system has the latest package versions installed, or before checking whether enabled repositories have
+    accessible URLs.
+    """
+    # We are using run_subprocess here as an alternative to call_yum_cmd
+    # which doesn't apply the correct --enablerepos option because at this point
+    # the tool didn't initialize the necessary functions in SystemInfo.
+    # The viable solution was calling the yum command as a subprocess manually
+    # instead of using that function wrapper.
+    output, ret_code = utils.run_subprocess(("yum", "clean", "metadata", "--quiet"), print_output=False)
+    loggerinst.debug("Output of yum clean metadata:\n%s" % output)
+
+    if ret_code != 0:
+        loggerinst.warning("Failed to clean yum metadata:\n%s" % output)
```

Comment:
Wondering if we should raise an `critical` here instead of warning. I would imagine that a broken metadata for some reason could be bad for the conversion, as the cache would have mixed states?

Reply:
Let me change it to a critical

Reply:
We don't know all the cases where yum returns non-0 code.
For example when you disable all repos (`yum-config-manager --disable "*"`) then `yum clean metadata` returns 1. But is it something we should halt the conversion on? I don't think so. Having no repos enabled is a strange state but it might be valid from the customer's POV and we shouldn't block the conversion on that.

Reply:
Noted